golang/cosmos/proto/agoric/vstorage/query.proto

```diff
@@ -37,6 +44,51 @@ message QueryDataResponse {
   ];
 }
 
+// QueryCapDataRequest contains a path and formatting configuration.
+message QueryCapDataRequest {
+  string path = 1 [
+    (gogoproto.jsontag)    = "path",
+    (gogoproto.moretags)   = "yaml:\"path\""
+  ];
+  // mediaType must be an actual media type in the registry at
+  // https://www.iana.org/assignments/media-types/media-types.xhtml
+  // or a special value that does not conflict with the media type syntax.
+  // The only valid value is "JSON Lines", which is also the default.
+  string media_type = 2 [
+    (gogoproto.jsontag)    = "mediaType",
+    (gogoproto.moretags)   = "yaml:\"mediaType\""
+  ];
+  // itemFormat, if present, must be the special value "flat" to indicate that
+  // the deep structure of each item should be flattened into a single level
+  // with kebab-case keys (e.g., `{ "metrics": { "min": 0, "max": 88 } }` as
+  // `{ "metrics-min": 0, "metrics-max": 88 }`).
+  string item_format = 3 [
+    (gogoproto.jsontag)    = "itemFormat",
+    (gogoproto.moretags)   = "yaml:\"itemFormat\""
+  ];
+  // remotableValueFormat indicates how to transform references to opaque but
+  // distinguishable Remotables into readable embedded representations.
+  // * "object" represents each Remotable as an `{ id, allegedName }` object.
+  // * "string" represents each Remotable as a bracketed string such as `[Alleged: IST brand {}]`.
```

Comment:
this is important documentation. is it published somewhere using godoc ?

Reply:
Not that I know of (as mentioned in the PR summary, we don't seem to have any documentation of RPC endpoints). Does anyone have a place where they would expect to find it?

Reply:
> we don't seem to have any documentation of RPC endpoints

It's far from perfect, but we have >0 for inter protocol RPC stuff: https://github.com/Agoric/agoric-sdk/tree/master/packages/inter-protocol#reading-data-off-chain

As to the rest, yes, it's been a hole in our docs for some time; I put what notes I learn in...

 - [Cosmos / tendermint RPC doc links · Issue \#657 · Agoric/documentation](https://github.com/Agoric/documentation/issues/657)

In some sense, this is a different issue, since that issue was about documentation of the form "Agoric stuff is based on cosmos-sdk; go read XYZ to find out how the lower level works." It turns out that there's no obvious candidate for XYZ, btw.

> Does anyone have a place where they would expect to find it?

The inter-protocol precedent suggests a readme under `golang/cosmos/x/vstorage`

Reply:
Added golang/cosmos/x/vstorage

Reply:
Also, it turns out that the documentation _is_ available via godoc of the generated pb.go.